girder/models/item.py

```diff
@@ -304,9 +304,23 @@ def setMetadata(self, item, metadata):
         item['meta'].update(six.viewitems(metadata))
 
         # Remove metadata fields that were set to null (use items in py3)
-        toDelete = [k for k, v in six.viewitems(item['meta']) if v is None]
-        for key in toDelete:
-            del item['meta'][key]
+        if not allowNull:
+            toDelete = [k for k, v in six.viewitems(metadata) if v is None]
+            for key in toDelete:
+                del item['meta'][key]
+
+        item['updated'] = datetime.datetime.utcnow()
+
+        # Validate and save the item
+        return self.save(item)
+
+    def deleteMetadata(self, item, fields):
+        if 'meta' not in item:
+            item['meta'] = {}
+
+        for field in fields:
+            if field in item['meta']:
+                del item['meta'][field]
```

Comment:
Also, the deletion logic in `setMetadata()` also uses `del`. Should I change that to `pop()` as well?